stm32_base/hal/fat_sd_stm32/fatfs/ff.c

```diff
@@ -2877,12 +2877,15 @@ FRESULT f_close (
 )
 {
     FRESULT res;
-
+    int res_test;
 
 #if !_FS_READONLY
     res = f_sync(fp);					/* Flush cached data */
-    if (res == FR_OK)
+    res_test = (res == FR_OK);
+#else
+    res_test = 1;

```

Comment:
Instead of having an `#else` here why not simply initialize res_test to = 1.  Saves you a couple lines of code.